### ISSUE TITLE
Autoscaling doesn't collect metric between different revisions correctly

### ISSUE THREAD

**Issue by StrongMonkey** [rancher/rio#344]
Rio version: v0.2.0
Steps to reproduce:
```
rio run -p 8080/http --name demo monachus/rancher-demo:yellow
rio scale default/demo=1-5
rio stage --image=monachus/rancher-demo:pink default/demo:v1
rio weight --rollout-interval 2 default/demo:v1=25
rio promote --rollout-interval 2 default/demo:v1
```
Create two revisions, each one can be scaled from 1 to 5. Send traffic to v0. Promoting v1 revision. Traffic should all be sent from v0 to v1. Noticed that autoscaler doesn't collect the correct info and don't scale v1 correspondingly.

Thanks for reporting this bug on slack by @harutyundermenjyan

**Comment by StrongMonkey** [comment]
Appears to be a regression when refactoring knative/serving v0.6.1 and envoy metric scraper.

**Comment by StrongMonkey** [comment]
Fixed in v0.3.0-rc2
